virttest/utils_libvirt/libvirt_disk.py

```diff
@@ -40,3 +43,40 @@ def create_disk(disk_type, path=None, size="500M", disk_format="raw", extra='',
     else:
         return libvirt.create_local_disk(disk_type, path=path, size=size,
                                          disk_format=disk_format, extra=extra)
+
+
+def create_primitive_disk(type_name, disk_device, device_target, device_bus,
+                          device_format, disk_src_dict, disk_auth_dict):
+    """
+    Creates primitive disk xml
+
+    :param type_name: disk type
+    :param disk_device: disk device
+    :param device_target: target device
+    :param device_bus: device bus
+    :param device_format: device format
+    :param disk_src_dict: disk source, dict format like below
+           disk_src_dict = {"attrs": {"protocol": "rbd",
+                                   "name": ceph_disk_name},
+                            "hosts":  [{"name": ceph_mon_ip,
+                                     "port": ceph_host_port}]}
+    :param disk_auth_dict: disk auth information, dict format like below
+           disk_auth_dict = {"auth_user": ceph_auth_user,
+                             "secret_type": auth_sec_usage_type,
+                             "secret_uuid": auth_sec_uuid}
```

Comment:
`disk_auth_dict` and `disk_src_dict` have `ceph_` parameter names. Is this a ceph specific function? Should the function maybe better be called `create_basic_ceph_disk_xml`?

If not, please change the names.

Reply:
function has been changed to create_primitive_disk_xml.
This is more of general disk generation, disk_auth_dict given here is just one demo how it looks like